### PR TITLE
[Agent v2.17] Stac 16209 open telemetry 1 1

### PR DESCRIPTION
**Pull request by MelcomVE**

### CODE REVIEW

File: configure/opentelemetry/manual-instrumentation/mappings.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+### Service Type
+  - `Key`
+    - service.type
+  - `Expected`
+    - This field can be any string value
+  - `Example`
+    - AWS RDS
+  - `Description`
+    - This will be the service displayed under your trace perspective for a specific trace.
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/mappings.md:120]: we should mention it what the user can use it for

File: configure/opentelemetry/manual-instrumentation/relations.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+---
+description: StackState Self-hosted v5.0.x
+---
+
+# Spans Relations
+
+Component and Span relations show the relationship between StackState components, for example, the image below shows an example of a relation running from the following to the following components:
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/relations.md:7]: `from the following to the following...` ? :thinking:

File: configure/opentelemetry/manual-instrumentation/merging.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
+---
+description: StackState Self-hosted v5.0.x
+---
+
+# Merging with StackState components
+Merging components allows you to do any of the following things
+- Add extra attributes into pre-existing StackState components.
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/merging.md:7]: all component are StackState components... 

maybe call it consistently pre-existing component ?
this is in several places, search for `StackState component` and change them all

**Review by TizianoPerrucci — CHANGES_REQUESTED** [review]

File: configure/opentelemetry/what-is-opentelemetry-instrumentation.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+## What is a span inside a trace
+A span represents each unit of work that has been done within the lifespan of the trace.
+A request's whole lifecycle, from creation to fulfillment, is represented by a trace.
+
+For example, in the trace captured below, three events occurred.
+
+The first was the Lambda that we monitored with OpenTelemetry, as seen in the screenshot below; this is the parent span.
+
+![trace span example c](../../.gitbook/assets/otel_example_trace_span_c.png)
```
[the remark this answers is not in the document]
Comment (MelcomVE) [reply configure/opentelemetry/what-is-opentelemetry-instrumentation.md:32]: Definitely forgot to update those

File: configure/opentelemetry/manual-instrumentation/instrumentation-examples.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+// Adding attributes to the child span
+childSpan.setAttribute('trace.perspective.name', 'RDS Table: Users');
+childSpan.setAttribute('service.name', 'RDS Table');
+childSpan.setAttribute('service.type', 'Database Tables');
+childSpan.setAttribute('service.identifier', 'rds:database:table:users');
+childSpan.setAttribute('resource.name', 'AWS RDS');
+
+// Closing the spans in order
+// You need to close the spans in the opposite order in which you opended them
```
[the remark this answers is not in the document]
Comment (MelcomVE) [reply configure/opentelemetry/manual-instrumentation/instrumentation-examples.md:125]: Added a comment to what will happen

File: configure/opentelemetry/manual-instrumentation/mappings.md (hunk 1 above)
Comment (MelcomVE) [reply configure/opentelemetry/manual-instrumentation/mappings.md:138]: I'm unable to find any functionality for this key except enhancing the data you can see in a Trace.

File: configure/opentelemetry/manual-instrumentation/instrumentation-examples.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+Example of how the parent component will look like if you create the with the above code
+
+![Parent Component after StackState received the trace](../../../.gitbook/assets/otel_example_parent.png)
+
+---
+
+### Child Span
+Now let's create a span that will have a relation with the parent span. Thus meaning if anything is wrong with the child span then the error will propagate up into the parent.
+
+Let's use a Table from the non exist Database as a child, this allows us to add a critical health state on the child if a write event to this table failed.
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/instrumentation-examples.md:156]: `write event` -> `write operation`

File: configure/opentelemetry/manual-instrumentation/merging.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+---
+
+# Merging with Pre-Existing components
+Merging components allows you to do any of the following things
+- Add extra attributes into pre-existing components.
+- Create custom relations for pre-existing components. 
+- Allows you to add a component relation to another propagating health in different ways (You can read more on the [span health state](/configure/opentelemetry/manual-instrumentation/health.md) page)
+
+## Important to know when merging
+When you merge your custom instrumentation with a StackState component, it might seem that your component disappeared; it did not.
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/merging.md:12]: `pre-existing` StackState component ?

File: configure/opentelemetry/manual-instrumentation/merging.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+The component you are merging with will inherit all the properties, health, and relations. This means that
+yes, you do not see your component anymore, but that's because the component you merged with will now act as the
+original component it initially appeared as, and the component you created.
+
+## Merging inheritance
+When you merge your component with a StackState component, the StackState component will contain all the labels, telemetry, and health from your original component.
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/merging.md:18]: ... with a `pre-existing` StackState component, the merged component ... ?

File: configure/opentelemetry/manual-instrumentation/merging.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+![Topology Perspective - Labels and Identifiers After Merge](../../../.gitbook/assets/merging_attempt.png)
+
+This is the same for the health checks and telemetry on the right side.
+
+![Topology Perspective - Component Merged Health State](../../../.gitbook/assets/merge_attempt_health.png)
+
+Now let's see the result after merging our child component with the pre-existing StackState component.
+
+Let's look at the `identifiers` and `labels` again. As you can see in the image below the
+`identifiers` stayed the same but the `labels` merged, This StackState component now contain the values from both.
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/merging.md:39]: `merged` component ?

File: configure/opentelemetry/manual-instrumentation/mappings.md [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+If you run you StackState Agent in debug mode then you should receive a message about an unknown instrumentation and the name that was passed to it
+
+## Span mapping requirements - Summary
+Below is a table with a summary of all the span keys that's required
+
+You need to include **ALL** the keys below when creating a span as they all need to be provided before the component will appear on your StackState instance.
+
+| ***Key***                       |  ***Type***  | ***Required***  | ***Allowed Value*** | ***Example***                |
+|:--------------------------------|:------------:|:---------------:|:-------------------:|:-----------------------------|
+| **trace.perspective.name**      |   `string`   |     **yes**     |     Any string      | AWS RDS: Hello World Database|
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/mappings.md:31]: the example value does not the example below, should be `RDS Database: Perspective Name`

File: configure/opentelemetry/manual-instrumentation/mappings.md [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+| **service.name**                |   `string`   |     **yes**     |     Any string      | AWS RDS: Database            |
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/mappings.md:32]: the example value does not the example below, should be `RDS Database: Service Name`

File: configure/opentelemetry/manual-instrumentation/mappings.md [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+| **resource.name**               |   `string`   |     **yes**     |     Any string      | Database                     |
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/mappings.md:33]: the example value does not the example below, should be `Database`
also to keep the same order as the example move this resource.name row after service.identifier

File: configure/opentelemetry/manual-instrumentation/mappings.md [hunk 13]
```diff
@@ -0,0 +1,247 @@
...
+| **service.type**                |   `string`   |     **yes**     |     Any string      | AWS RDS                      |
+| **service.identifier**          |   `string`   |     **yes**     |     Any string      | aws:rds:database:hello-world |
+| **http.status_code**            |   `number`   |     **no**      |     HTML status     | 200                          |
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/mappings.md:36]: `HTML status` -> `HTTP status`

File: configure/opentelemetry/manual-instrumentation/mappings.md [hunk 14]
```diff
@@ -0,0 +1,247 @@
...
+We will show how the above is implemented when we get to the code examples page. For now, it is good to know that if the above does not match what the StackState Agent is expecting then it will not be displayed on StackState. 
+
+If you run you StackState Agent in debug mode then you should receive a message about an unknown instrumentation and the name that was passed to it
+
+## Span mapping requirements - Summary
+Below is a table with a summary of all the span keys that's required
+
+You need to include **ALL** the keys below when creating a span as they all need to be provided before the component will appear on your StackState instance.
+
+| ***Key***                       |  ***Type***  | ***Required***  | ***Allowed Value*** | ***Example***                |
```
Comment (TizianoPerrucci) [inline configure/opentelemetry/manual-instrumentation/mappings.md:29]: can we link each key example to its section ?

**Review by TizianoPerrucci — CHANGES_REQUESTED** [review]

**Pull request merged**